From 023974b2531c63d5f2b12fb98e7ed1f806302267 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 16 Oct 2025 06:31:01 +0800
Subject: [PATCH 1/2] CMakeLists.txt: use external gtest, fallback to fetching
 the source (#288)

See: https://github.com/greg7mdp/parallel-hashmap/issues/287
(cherry picked from commit 8bd66dc8ed98ba0e941864edf4d3670e60543f8b)
---
 CMakeLists.txt | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef0d46e..57625fc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,12 +96,17 @@ endif()
 if (PHMAP_BUILD_TESTS)
 
     if (NOT PHMAP_GTEST_LIBS)
-        include(cmake/DownloadGTest.cmake)
-
-        check_target(gtest)
-        check_target(gtest_main)
-        check_target(gmock)
-        set(PHMAP_GTEST_LIBS gmock_main)
+        find_package(GTest CONFIG)
+        if (GTest_FOUND)
+            set(PHMAP_GTEST_LIBS GTest::gmock_main)
+        else()
+            include(cmake/DownloadGTest.cmake)
+
+            check_target(gtest)
+            check_target(gtest_main)
+            check_target(gmock)
+            set(PHMAP_GTEST_LIBS gmock_main)
+        endif()
     endif()
 
     enable_testing()
-- 
2.53.0

